src/lib/operators/join_sort_merge.cpp

```diff
@@ -602,14 +626,15 @@ class JoinSortMerge::JoinSortMergeImpl : public AbstractReadOnlyOperatorImpl {
   /**
   * Determines the largest value in a sorted materialized table.
   **/
-  T& _table_max_value(std::unique_ptr<MaterializedSegmentList<T>>& sorted_table) {
+  const T& _table_max_value(const MaterializedSegmentList<T>& sorted_table) {
     DebugAssert(_primary_predicate_condition != PredicateCondition::Equals,
                 "The table needs to be sorted for _table_max_value() which is only the case in the non-equi case");
-    DebugAssert(!sorted_table->empty(), "Sorted table is empty");
+    DebugAssert(!sorted_table.empty(), "Sorted table is empty");
 
-    for (size_t partition_id = sorted_table->size() - 1; partition_id < sorted_table->size(); --partition_id) {
-      if (!(*sorted_table)[partition_id]->empty()) {
-        return (*sorted_table)[partition_id]->back().value;
+    const auto sorted_table_size = sorted_table.size();
+    for (auto partition_id = sorted_table_size - 1; partition_id < sorted_table_size; --partition_id) {
```

Comment:
Does this loop variable underflow since `partition_id` is only decreased and already smaller than `sorted_table_size`?

Reply:
vector->size returns a size_t which should be an unsigned integer. The loop stops once the value underflows and is greater than sorted_table_size again. #MakePartitionIdGreatAgain.

Reply:
I don't think its great coding style, it should be more explicit tbh.

Reply:
Ahhh, thanks a lot. Absolutely haven't seen this issue, @Bensk1. When I read your comment I was sure it underflow, but @ArneMayer should be right. Both with 0 - 1 and with the coding style.

Will definitely change it. Whuuaaarghh ...

Reply:
Changed to
```
for (auto partition_iter = sorted_table.rbegin(); partition_iter != sorted_table.rend(); ++partition_iter) {
      if (!partition_iter->empty()) {
        return partition_iter->back().value;
      }
    }
```

Reply:
much better!